src/Symfony/Component/Runtime/BaseRuntime.php

```diff
@@ -0,0 +1,179 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime;
+
+/**
+ * A runtime to do bare-metal PHP without using superglobals.
+ *
+ * One option named "debug" is supported; it toggles displaying errors.
+ *
+ * The app-closure returned by the entry script must return either:
+ * - "string" to echo the response content, or
+ * - "int" to set the exit status code.
+ *
+ * The app-closure can declare arguments among either:
+ * - "array $context" to get a local array similar to $_SERVER;
+ * - "array $argv" to get the command line arguments when running on the CLI;
+ * - "array $request" to get a local array with keys "query", "data", "files" and
+ *   "session", which map to $_GET, $_POST, $FILES and &$_SESSION respectively.
+ *
+ * The runtime sets up a strict error handler that throws
+ * exceptions when a PHP warning/notice is raised.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class BaseRuntime implements RuntimeInterface
+{
+    private $debug;
+
+    public function __construct(array $options = [])
+    {
+        $this->debug = $options['debug'] ?? true;
+        error_reporting(-1);
+
+        if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg'], true)) {
+            ini_set('display_errors', $this->debug);
+        } elseif (!filter_var(ini_get('log_errors'), FILTER_VALIDATE_BOOLEAN) || ini_get('error_log')) {
+            // CLI - display errors only if they're not already logged to STDERR
+            ini_set('display_errors', 1);
+        }
+
+        if (0 <= ini_get('zend.assertions')) {
+            ini_set('zend.assertions', 1);
+            ini_set('assert.active', $this->debug);
+            ini_set('assert.bail', 0);
+            ini_set('assert.warning', 0);
+            ini_set('assert.exception', 1);
+        }
+
+        set_error_handler(static function ($type, $msg, $file, $line) {
+            if ((E_DEPRECATED | E_USER_DEPRECATED) & $type) {
+                return true;
+            }
+
+            if ((error_reporting() | E_ERROR | E_RECOVERABLE_ERROR | E_PARSE | E_CORE_ERROR | E_COMPILE_ERROR | E_USER_ERROR) & $type) {
+                throw new \ErrorException($msg, 0, $type, $file, $line);
+            }
+
+            return false;
+        });
+
+        set_exception_handler(static function ($e) {
+            echo "<pre>\n$e\n</pre>\n";
+        });
+    }
+
+    public function resolve(\Closure $app): \Closure
+    {
+        $arguments = [];
+        $function = new \ReflectionFunction($app);
+
+        try {
+            foreach ($function->getParameters() as $parameter) {
+                $arguments[] = $this->getArgument($parameter, $parameter->getType());
+            }
+        } catch (\InvalidArgumentException $e) {
+            if (!$parameter->isOptional()) {
+                throw $e;
+            }
+        }
+
+        $returnType = $function->getReturnType();
+        $returnType = $returnType instanceof \ReflectionNamedType ? $returnType->getName() : '';
```

Comment:
You are using the same var to store an object or a string. This is not optimal for performance

Reply:
I just ran some bench and there is zero difference between using another name or not.